src/main/ruby/truffleruby/core/string.rb

```diff
@@ -1490,12 +1490,22 @@ def rindex(sub, finish=undefined)
   end
 
   def start_with?(*prefixes)
+    binding = TrufflePrimitive.caller_binding
+
     prefixes.each do |original_prefix|
-      prefix = Truffle::Type.rb_check_convert_type original_prefix, String, :to_str
-      unless prefix
-        raise TypeError, "no implicit conversion of #{original_prefix.class} into String"
+      case original_prefix
+      when Regexp
+        Truffle::Type.compatible_encoding(self, original_prefix)
+        match_data = original_prefix.match_onwards(self, 0, true)
```

Comment:
@aardvark179 Can you double check this sounds right for `String#start_with?(Regexp)`?
It sounds good to me, but we have quite a few not-so-clearly-named Regexp matching primitives.

Reply:
Yes, that looks like the correct primitive with the correct arguments for doing this.